src/driver-sinowealth.c

```diff
@@ -1527,7 +1529,7 @@ sinowealth_init_profile(struct ratbag_device *device)
 	 */
 	struct sinowealth_config_report *config = &drv_data->configs[0];
 
-	char fw_version[4];
+	char fw_version[SINOWEALTH_FW_VERSION_LEN];
```

Comment:
```c
char fw_version[SINOWEALTH_FW_VERSION_LEN + 1] = {0};
```
and bam, you have a guaranteed null-terminated string and don't need to worry about the exact length anymore.

Reply:
Super nice!